src/components/session-list.tsx

```diff
@@ -0,0 +1,147 @@
+        setLoading(false);
+      }
+    }
+
+    fetchSessions();
+  }, []);
+
+  const handleGenerateNewCode = () => {
+    router.push("/generate"); // Navigate to the generate code page
+  };
+
+  if (loading) {
+    return (
+      <div className="flex justify-center items-center h-screen">
+        Loading sessions...
+      </div>
+    );
+  }
+
+  if (error) {
+    return <div className="text-red-500 text-center">{error}</div>;
+  }
+
+  return (
+    <>
+      <Card className="w-full max-w-lg mx-auto mt-8">
+        <CardHeader>
+          <CardTitle className="text-2xl font-bold text-center">
+            Ongoing Karaoke Sessions
+          </CardTitle>
+        </CardHeader>
+        <CardContent>
+          {sessions.length === 0 ? (
+            <p className="text-center text-gray-500">
+              No active sessions at the moment.
+            </p>
+          ) : (
+            <ul className="space-y-2">
+              {sessions.map((session) => (
+                <li key={session.id}>
+                  <div
+                    className="flex justify-between items-center p-2 bg-secondary rounded-md hover:bg-primary hover:text-primary-foreground transition-colors duration-200 cursor-pointer"
+                    onClick={() => router.push(`/session/${session.id}`)}
+                  >
+                    <div>
+                      <span className="font-medium">{session.sessionName}</span>
+                      <div className="text-sm text-muted-foreground">
+                        {session.queueCount}{" "}
+                        {session.queueCount === 1 ? "person" : "people"} in
+                        queue
+                      </div>
+                    </div>
+                    <Badge variant="outline" className="ml-2">
+                      {session.id}
+                    </Badge>
+                  </div>
+                </li>
+              ))}
+            </ul>
+          )}
+          <div className="text-center mt-4 flex-grow">
+            <Button className="flex-grow" onClick={handleGenerateNewCode}>
+              Create a new session code
+            </Button>
+          </div>
+        </CardContent>
+      </Card>
+      <Button
+        className="fixed bottom-5 left-5 rounded-lg p-4 bg-primary text-primary-foreground shadow-lg hover:bg-primary-hover"
+        onClick={() => router.push("/")}
+      >
+        <ArrowLeftIcon className="w-6 h-6" />
+      </Button>
+    </>
+  );
+}
+
+function ArrowLeftIcon(
+  props: JSX.IntrinsicAttributes & SVGProps<SVGSVGElement>
+) {
+  return (
+    <svg
+      {...props}
+      xmlns="http://www.w3.org/2000/svg"
+      width="24"
+      height="24"
+      viewBox="0 0 24 24"
+      fill="none"
+      stroke="currentColor"
+      strokeWidth="2"
+      strokeLinecap="round"
+      strokeLinejoin="round"
+    >
+      <path d="M19 12H5" />
+      <path d="m12 19-7-7 7-7" />
+    </svg>
+  );
+}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `ArrowLeftIcon` component is defined inside the `SessionList` component file. For better modularity and reusability, consider moving this component to its own file under a directory dedicated to icons. This way, it can be easily reused across different components and files.


<!-- This is an auto-generated comment by OSS CodeRabbit -->